nft_ingester/src/index_syncronizer.rs

```diff
@@ -210,26 +252,35 @@ where
             1,
             false,
         ));
-        let task = tokio::spawn(async move {
-            temp_syncronizer
-                .run(&local_rx, -1, tokio::time::Duration::from_millis(100))
-                .await;
-        });
 
-        self.dump_sync(last_included_rocks_key.as_slice(), rx)
+        match asset_type {
+            AssetType::NonFungible => {
+                temp_syncronizer
+                    .nft_run(&local_rx, -1, tokio::time::Duration::from_millis(100))
+                    .await
+            }
+            AssetType::Fungible => {
+                temp_syncronizer
+                    .fungible_run(&local_rx, -1, tokio::time::Duration::from_millis(100))
+                    .await
+            }
+        }
+        self.dump_sync(last_included_rocks_key.as_slice(), rx, asset_type)
             .await?;
 
         tx.send(()).map_err(|e| e.to_string())?;
-        task.await.map_err(|e| e.to_string())?;
+
         // now we can copy temp storage to the main storage
         temp_storage.copy_to_main().await?;
+
         Ok(())
     }
 
     async fn dump_sync(
         &self,
         last_included_rocks_key: &[u8],
         rx: &tokio::sync::broadcast::Receiver<()>,
+        asset_type: AssetType,
     ) -> Result<(), IngesterError> {
         let path = std::path::Path::new(self.dump_path.as_str());
         tracing::info!("Dumping the primary storage to {}", self.dump_path);
```

Comment:
Below it completes a dump of both fungibles and nfts, which is a multi-hour process, that should not be run twice, but instead be separated into 2 dumps